src/main/java/jp/ikedam/jenkins/plugins/extensible_choice_parameter/DatabaseChoiceListProvider.java

```diff
@@ -0,0 +1,419 @@
+/*
+ * The MIT License
+ * 
+ * Copyright (c) 2013 IKEDA Yasuyuki
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package jp.ikedam.jenkins.plugins.extensible_choice_parameter;
+
+import hudson.Extension;
+import hudson.model.Descriptor;
+import hudson.util.FormValidation;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.Serializable;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Scanner;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.apache.commons.lang.StringUtils;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.QueryParameter;
+
+/**
+ * Create a choice list by querying a specific column of a table in a database.
+ * 
+ * @author Thibault Duchateau
+ */
+public class DatabaseChoiceListProvider extends ChoiceListProvider implements Serializable
+{
+	private static final long serialVersionUID = 8488443485003760275L;
+        
+    private static final Logger LOGGER = Logger.getLogger(DatabaseChoiceListProvider.class.getName());
+    
+    private String jdbcUrl;
+    private String dbName;
+    private String dbTable;
+    private String dbColumn;
+    private String dbUsername;
+    private String dbPassword;
+    private String jdbcDriver;
+    private String fallbackFilePath;
+    
+	public String getDbUsername() {
```

Comment:
Inconsistent indentation.
Please don't use tab letters for indentations. Use only blank letters for that.